contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+        IVault(vaultAddress).burnForStrategy(OETHbBalance);
+    }
+
+    function updateUnderlyingAssets() internal {
+        if (tokenId == 0) {
+            underlyingAssets = 0;
+        } else {
+            (uint128 liquidity,,) = _getPositionInfo();
+
+            /**
+             * Our net value represent the smallest amount of tokens we are able to extract from the position
+             * given our liquidity.
+             *
+             * The least amount of tokens extraditable from the position is where the active trading price is
+             * at the ticker 0 meaning the pool is offering 1:1 trades between WETH & OETHb. At that moment the pool 
+             * consists completely of OETHb and no WETH.
+             * 
+             * The more swaps from WETH -> OETHb happen on the pool the more the price starts to move towards the -1
+             * ticker making OETHb (priced in WETH) more expensive.
+             */
+            (uint256 wethAmount, uint256 OETHbAmount) = helper.getAmountsForLiquidity(
+                sqrtRatioX96TickClosestToParity, // sqrtRatioX96
+                sqrtRatioX96Tick0,               // sqrtRatioAX96
+                sqrtRatioX96Tick1,               // sqrtRatioBX96
+                liquidity
+            );
+
+            require(wethAmount == 0, "Non zero wethAmount");
+            underlyingAssets = OETHbAmount;
+        }
+
+    }
+
+    /**
+     * @notice Deposit all supported assets in this strategy contract to the platform
+     */
+    function depositAll()
+        external
+        override
+        onlyVault
+        nonReentrant 
+    {
+        _deposit(WETH, IERC20(WETH).balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Withdraw an `amount` of assets from the platform and
+     * send to the `_recipient`.
+     * @param _recipient         Address to which the asset should be sent
+     * @param _asset             Address of the asset
+     * @param _amount            Units of asset to withdraw
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) 
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        require(_asset == WETH, "Unsupported asset");
+        require(_recipient == vaultAddress, "Only withdraw to vault allowed");
+
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (wethBalance < _amount) {
+            uint256 additionalWETHRequired = _amount - wethBalance;
+            (uint256 WETHInThePool,) = getPositionPrincipal();
+            
+            if(WETHInThePool < additionalWETHRequired) {
+                revert NotEnoughWethLiquidity(WETHInThePool, additionalWETHRequired);
+            }
+
+            uint256 shareOfWethToRemove = additionalWETHRequired.divPrecisely(WETHInThePool) + 1;
+            _removePartialLiquidity(shareOfWethToRemove);
+        }
+
```

Comment:
Will this method leave dust WETH?

Reply:
`shareOfWethToRemove` is in bps, not absolute value. And it corresponds to the amount of liquidity to remove from the pool to get `x` amount of WETH (and equivalent OETHb). So, it's most likely that there will be a few weis after this